### PR TITLE
remove manual upload port and switch to autodiscovery in platformio

### PR DESCRIPTION
**Pull request by blakadder**
## Description:
Remove manual port entry as default and use Platformio port discovery as default.

Needed setting`common.upload_port` to a MAC default so the CI doesn't break

Option to set upload port is now in `platformio_override.ini` to keep user configurations out of default files. Being empty enables port discovery on my Windows system.

Tested on different boards. Does not work if multiple flashable devices are connected.

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.5
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by Jason2866** [comment]
CI Fails on Mac

File: platformio_override_sample.ini [hunk 1]
```diff
@@ -48,6 +48,10 @@ default_envs            =
 
 
 [env]
+; *** Define serial port used for erasing/flashing/terminal
+upload_port                 = COM10
+monitor_speed               = 115200
```
Comment (Jason2866) [inline platformio_override_sample.ini:53]: Why override `monitor_speed` ?

**Comment by blakadder** [comment]
doesn't matter, mac is not playing along as always

File: platformio.ini [hunk 2]
```diff
@@ -35,10 +35,10 @@ custom_unpack_dir           = unpacked_littlefs
 build_unflags               = ${core.build_unflags}
 build_flags                 = ${core.build_flags}
 monitor_speed               = 115200
-monitor_port                = COM5
+; monitor_port                = COM5
```
Comment (Jason2866) [inline platformio.ini:38]: Why not remove?

Comment (blakadder) [reply platformio.ini:38]: i dunno, cause mac might have some objections

File: platformio_override_sample.ini [hunk 3]
```diff
@@ -113,7 +113,6 @@ lib_extra_dirs          = ${library.lib_extra_dirs}
 ;board_build.partitions  = partitions/esp32_partition_app2944k_fs2M.csv
 monitor_speed           = 115200
 ; *** Serial port used for erasing/flashing the ESP32
-;upload_port             = ${common.upload_port}
 upload_port             = COM4
```
Comment (Jason2866) [inline platformio_override_sample.ini:116]: Not consistent. Should be `upload_port = `

File: platformio.ini (hunk 2 above)
Comment (Jason2866) [reply platformio.ini:38]: It is only a comment.

**Pull request merged**